### PR TITLE
refactor button, link and input to have tooltip on disabled behavior from the shared lib and input-text use the form input hoc

### LINKED ISSUE #170: refactor button, link and text input to use the new disabledTooltip hoc
This is the followup for #166 to replace the duplicated disabled tooltip code with the imported lib, remove the relevant styles and revisit written tests

### LINKED ISSUE #176: refactor text input component to use the form input hoc
after #165 is done, the existing text input  component should be refactored to use the new hoc, and we need to revisit/delete old tests that are duplicate with the ones in the hoc package

### PR DESCRIPTION
**Pull request by fczuardi**
- affects: ``@crave/farmblocks-button``, ``@crave/farmblocks-input-text``, ``@crave/farmblocks-link``
- [x] closes #170
- [x] closes #176

### CODE REVIEW

**Comment by fczuardi** [comment]
This is good for reviewing again :)

File: packages/link/src/components/Link.js [hunk 1]
```diff
@@ -1,93 +1,66 @@
...
-          isVisible={this.state.showTooltip}
-          align={props.tooltipAlign}
-        />
-      </Container>
-    );
-  }
-
-  onClick(e) {
-    e.stopPropagation();
-    return this.props.onClick(e);
```
[the remark this answers is not in the document]
Comment (alcferreira) [reply packages/link/src/components/Link.js]: Anyway i'll try to test this situation with the `canary` version

[the remark this answers is not in the document]
Comment (fczuardi) [reply packages/link/src/components/Link.js]: if you pass an onClick function to the Link component, it should be forwarded to the underlining A link... but yeah, if you were relying on the stopPropagation wrapper the previous behaviour will change and we should document this as a breaking change to trigger a major version bump in the link component

[the remark this answers is not in the document]
Comment (fczuardi) [reply packages/link/src/components/Link.js]: just checked and the breaking change notice is there, so commit 598811363b34faf7b37b0194038da2e3b466abae will trigger a major version bump on farmblocks-link

[the remark this answers is not in the document]
Comment (alcferreira) [reply packages/link/src/components/Link.js]: Yeah, my bad. Checked here and `Footer` component, that i was referring before is not relying on this wrapper.

File: packages/hoc-disabled-tooltip/src/disabledTooltip.js [hunk 2]
```diff
@@ -5,6 +5,15 @@ import Tooltip from "@crave/farmblocks-tooltip";
 
 import Wrapper from "./styledComponents/Wrapper";
 
+export const newPropTypes = {
```
Comment (alcferreira) [inline packages/hoc-disabled-tooltip/src/disabledTooltip.js]: hmmmm.... why `newPropTypes`? 
Can we change to something like `disabledTooltipProps`?

Comment (fczuardi) [reply packages/hoc-disabled-tooltip/src/disabledTooltip.js]: yes, your suggestion is better

**Comment by fczuardi** [comment]
changes addressed

**Comment by fczuardi** [comment]
rebased and incorporated the new rightIcon property in the button refactor... good for review

**Comment by alcferreira** [comment]
Ops... conflict warning ⚠️

**Comment by fczuardi** [comment]
snapshot needs update, I am on it

**Comment by fczuardi** [comment]
because the rightIcon Button PR got merged before this one, and then the dropdown, that uses that version of the Button

**Comment by fczuardi** [comment]
rebased

**Comment by alcferreira** [comment]
@vNakamura can you review this pr?

**Comment by vNakamura** [comment]
@alcferreira sure

File: packages/input-text/README.md [hunk 3]
```diff
@@ -73,8 +72,8 @@ the default native browser balloon and display the validation message provided
 by the browser using the farmblocks styling.
 
 We recomend however that you do back-end validation regardless of the browser
-constraint support, and then use the `invalid` and `validationMessages`
-properties to present the returned field errors of your server-side validations.
+constraint support, and then use the `validationMessages`
+propertiy to present the returned field errors of your server-side validations.
```
Comment (vNakamura) [inline packages/input-text/README.md]: Little mistype for `property`.

File: packages/input-text/README.md [hunk 4]
```diff
@@ -51,8 +51,7 @@ properties that provides enhanced functionality:
 | property           | type             | description                                                                                                                 |
 | ------------------ | ---------------- | --------------------------------------------------------------------------------------------------------------------------- |
 | label              | string           | a text to be used as label for the field                                                                                    |
-| invalid            | boolean          | if passed will style the field as a field that needs attention / corrections                                                |
-| validationMessages | array of strings | text messages to display if the invalid property is present / true                                                          |
+| validationMessages | array of strings | text messages to display if the invalid property is present / true if passed will style the field as a field that needs attention / corrections |
```
Comment (vNakamura) [inline packages/input-text/README.md]: The description mentions the `invalid` prop, but it was removed.
